File: Lowest Common Ancestor in BST/Leetcode_235.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        if (p->val > q->val) {
            TreeNode* tmp = p;
            p = q;
            q = tmp;
        }
        while (root->val < p->val || root->val > q->val){
            cout << root->val << endl;
            while(root->val < p->val) {root=root->right; cout << root->val << endl;}
            while(root->val > q->val) {root=root->left; cout << root->val << endl;}
        }
        return root;
    }
};

File: Lowest Common Ancestor in BST/Leetcode_235.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        if p.val > q.val: p, q = q, p
        while p.val>root.val or q.val<root.val:
            while p.val>root.val: root = root.right
            while q.val<root.val: root = root.left
        return root
